client/Android/FreeRDPCore/jni/android_freerdp.c

```diff
@@ -111,20 +111,44 @@ void android_begin_paint(rdpContext* context)
 
 void android_end_paint(rdpContext* context)
 {
+	int i;
+	int ninvalid;
+	HGDI_RGN cinvalid;
+	int x1, y1, x2, y2;
 	androidContext *ctx = (androidContext*)context;
 	rdpSettings* settings = context->instance->settings;
 	
-	DEBUG_ANDROID("ui_update");
-
 	assert(ctx);
 	assert(settings);
 	assert(context->instance);
 
-	DEBUG_ANDROID("width=%d, height=%d, bpp=%d", settings->DesktopWidth,
-			settings->DesktopHeight, settings->ColorDepth);
+	ninvalid = ctx->rdpCtx.gdi->primary->hdc->hwnd->ninvalid;
+	if (ninvalid == 0)
+	{
+		DEBUG_ANDROID("ui_update: ninvalid=%d", ninvalid);
+		return;
+	}
+
+	cinvalid = ctx->rdpCtx.gdi->primary->hdc->hwnd->cinvalid;
+
+	x1 = cinvalid[0].x;
+	y1 = cinvalid[0].x + cinvalid[0].y;
+	x2 = cinvalid[0].w;
```

Comment:
Yes. It's my mistake.
Should be:
x1 = cinvalid[0].x;
y1 = cinvalid[0].y;
x2 = cinvalid[0].x + cinvalid[0].w;
y2 = cinvalid[0].y + cinvalid[0].h;